lib/xypn_seo/seo_evaluation/evaluation.rb

```
@@ -0,0 +1,33 @@
+require 'rest-client'
+require 'json'
+
+class SiteEval
+
+  def call
+    evaluate_all_websites
+  end
+
+  private
+
+  def evaluate_all_websites
+    Website.all.each do | website |
+      puts "Evaluating #{website.url}"
+      website[:desktop_score] = overall_score(website.url, "desktop")
+      website[:mobile_score] = overall_score(website.url, "mobile")
+      website.save!
+    end
+  end
+
+  def overall_score(url, strategy)
+    begin
+      url = "https://www.googleapis.com/pagespeedonline/v1/runPagespeed?url=#{url}&key=#{ENV['PAGESPEED_KEY']}&prettyprint=false&strategy=#{strategy}"
+      response = RestClient.get url
+    rescue RestClient::BadRequest
+      0
```

We don't want to gobble up errors like this.  If we're rescuing an error, we should at least log it.  Trust me, you don't want to debug code that fails silently.  You may not want to tell the user about all the errors that happen, but you'll at least want to make the error's presence known.

Try something like this

```
rescue RestClient::BadRequest => e
  Rails.logger.error(e)
```

This will at least send the error message to the server logs.  We can also add e.backtrace or provide a custom message.

---

I still need to return an integer for the Website score columns else the scope query places nils at the highest value

---

Makes sense.  Most import thing here is that we're logging that something went wrong.